### PR TITLE
V8: Use infinite editor for list view settings (5021)

### LINKED ISSUE #5021: UX: List view configuration should open as an infinite editor.
I would like the List View-configuration (in DocType-configuration) to open as an infinite editor.

Today it looks like:
image

The reason for this is to ensure a better scope for the current editing phase. Today the "Save & publish"-button is placed in a way where it invites the user to confirm their changes by clicking it. But the save-feature of the list-view-configuration is an inline button further down the page. So we need to create a focus so the user doesn't see the "Save & Publish"-button.

By implementing this as infinite editing we will ensure that the users focus is only on the configuration of the list view and we will be able to make the "save"-button stay visible in bottom right corner always.

When implemented I would like Listview configuration to be visible in the "Permissions"-tab. Which I then want renamed to "Configurations".

All question and suggestions are very welcome.

### PR DESCRIPTION
**Pull request by nathanwoulfe**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes #5021 

### Description
Changes here update the existing list view settings user experience to use an infinite editor rather than the collapsing panel. I've tidied up the directive a little, and updated the property styles in the overlay to make everything more consistent and v8-flavoured.

Gif for viewing pleasure:

list-view

Have verified everything works as before.

### CODE REVIEW

**Comment by emmaburstow** [comment]
Oh lovely stuff, @nathanwoulfe! Excellent fix. We'll take a look at the work and shout if we need anything from you, as ever. 

Emma

**Comment by nathanwoulfe** [comment]
Cheers legend! 😁

File: src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js [hunk 1]
```diff
@@ -1,161 +1,143 @@
...
+            } 
 
-          });
+            /* ----------- LIST VIEW SETTINGS --------- */
+            const showSettingsOverlay = () => {
+                const overlay = {
+                    view: 'views/components/umb-list-view-settings-overlay.html',
+                    hideDescription: true,
+                    hideIcon: true,
+                    size: 'small',
```
Comment (skttl) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js]: I think it should be in the large infinite editor, that would give the settings some space to breathe.

Comment (nathanwoulfe) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js]: Hi @skttl, missed your comment - is there an inbetween size? The default (per infinite editor for a document type) might be too wide, something inbetween would be goldilocks-just-right. Can add an umb-editor--medium option easily enough

Comment (skttl) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js]: No, only large and small is supported. But maybe a PR with a medium setting would be approved? :)

Comment (nul800sebastiaan) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js]: Looks like I never replied to this one : https://github.com/umbraco/Umbraco-CMS/issues/5516 :-o

But yes, we'd be good with adding a medium!

Comment (nathanwoulfe) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js]: One medium coming up. Will keep it separate from this PR though.

If I update my local branch from upstream/v8/dev, will I end up with a heap of additional git history/commits? This branch was 8.1, so it's stale.

Comment (nul800sebastiaan) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js]: You will, but they'll be in your local v8/dev branch. Shouldn't be a problem! And you could even merge v8/dev into this PR branch as well.

Comment (nul800sebastiaan) [reply src/Umbraco.Web.UI.Client/src/common/directives/components/umblistviewsettings.directive.js]: I've accepted the `medium` PR and updated this PR. Also merged with v8/dev which went well.

**Comment by nul800sebastiaan** [comment]
Sorry for the delay @nathanwoulfe - I am loving the update and the implementation is a joy to work with. No more "double save button" that I (and everybody else) keep forgetting to press! 🎉

Thanks again!! ⭐️

**Comment by nielslyngsoe** [comment]
Thanks thanks thanks! 🤩

**Comment by nathanwoulfe** [comment]
@nul800sebastiaan @nielslyngsoe my pleasure!

**Pull request merged**